### DISCUSSION TITLE
get_events API-Endpoint /events

### DISCUSSION THREAD

**Discussion by EnricoBaivo** [NiceAesth/aiosu#163] [Ideas]
Would be great if you could add that API Endpoint to the Aiosu libary.
With this you get the Achievementsdata for a User and other relevant activitys. 

```JSON

{"achievement": {
                "icon_url": "https://assets.ppy.sh/medals/web/all-secret-youre-here-forever.png",
                "id": 302,
                "name": "You're Here Forever",
                "grouping": "Hush-Hush",
                "ordering": 2,
                "slug": "all-secret-youre-here-forever",
                "description": "We knew you'd be back.",
                "mode": null,
                "instructions": "<i>Touch grass.</i>"
            },
}
```
### JSON Response example
[EVENTS_dump_notimpl_JSON.txt](https://github.com/NiceAesth/aiosu/files/12431833/EVENTS_dump_notimpl_JSON.txt)

Aiosu already delivers the correct models for the response, it only misses the corresponding get_events method.

## Offical OSU WEB api v2 Doku: 

Get Events
[OAuth](https://osu.ppy.sh/docs/index.html#scope-oauth) [public](https://osu.ppy.sh/docs/index.html#scope-public)
Request
GET /events
Returns a collection of [Event](https://osu.ppy.sh/docs/index.html#event)s in order of creation time.
<html>
<body>
<form style="color: rgb(51, 51, 51); font-family: &quot;Open Sans&quot;, &quot;Helvetica Neue&quot;, Helvetica, Arial, sans-serif; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(234, 242, 246); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><h4 class="fancy-heading-panel" style="background-color: rgb(211, 211, 211); border-radius: 5px; margin: 2.5em 1136px 0.8em 25px; width: 1067.83px; padding: 0px 28px; font-family: &quot;Open Sans&quot;, &quot;Helvetica Neue&quot;, Helvetica, Arial, sans-serif; font-weight: 700; font-size: 10px; box-sizing: border-box; display: block; text-shadow: rgb(255, 255, 255) 0px 1px 0px;"><b style="font-weight: bolder;">Query Parameters</b></h4><div style="margin-right: 1136px; padding: 0px 28px; box-sizing: border-box; display: block; text-shadow: rgb(255, 255, 255) 0px 1px 0px; clear: unset;"><b style="font-weight: bolder; line-height: 2;"><code style="font-family: Consolas, Menlo, Monaco, &quot;Lucida Console&quot;, &quot;Liberation Mono&quot;, &quot;DejaVu Sans Mono&quot;, &quot;Bitstream Vera Sans Mono&quot;, &quot;Courier New&quot;, monospace, serif; font-size: 14px; line-height: 1.5; word-break: break-word; hyphens: auto; padding: 3px; border-radius: 3px; background-color: rgba(0, 0, 0, 0.05);">cursor_string</code></b>  <span> </span><small style="font-size: 11.2px;">string</small> <span> </span><i>optional</i><span> </span> <br><p style="line-height: 1.6; margin-top: 0px;">Parameter for pagination.</p></div><div style="margin-right: 1136px; padding: 0px 28px; box-sizing: border-box; display: block; text-shadow: rgb(255, 255, 255) 0px 1px 0px; clear: unset;"><b style="font-weight: bolder; line-height: 2;"><code style="font-family: Consolas, Menlo, Monaco, &quot;Lucida Console&quot;, &quot;Liberation Mono&quot;, &quot;DejaVu Sans Mono&quot;, &quot;Bitstream Vera Sans Mono&quot;, &quot;Courier New&quot;, monospace, serif; font-size: 14px; line-height: 1.5; word-break: break-word; hyphens: auto; padding: 3px; border-radius: 3px; background-color: rgba(0, 0, 0, 0.05);">sort</code></b>  <span> </span><small style="font-size: 11.2px;">string</small> <span> </span><i>optional</i><span> </span> <br><p style="line-height: 1.6; margin-top: 0px;">Sorting option. Valid values are<span> </span><code style="font-family: Consolas, Menlo, Monaco, &quot;Lucida Console&quot;, &quot;Liberation Mono&quot;, &quot;DejaVu Sans Mono&quot;, &quot;Bitstream Vera Sans Mono&quot;, &quot;Courier New&quot;, monospace, serif; font-size: 14px; line-height: 1.5; word-break: break-word; hyphens: auto; padding: 3px; border-radius: 3px; background-color: rgba(0, 0, 0, 0.05);">id_desc</code><span> </span>(default) and<span> </span><code style="font-family: Consolas, Menlo, Monaco, &quot;Lucida Console&quot;, &quot;Liberation Mono&quot;, &quot;DejaVu Sans Mono&quot;, &quot;Bitstream Vera Sans Mono&quot;, &quot;Courier New&quot;, monospace, serif; font-size: 14px; line-height: 1.5; word-break: break-word; hyphens: auto; padding: 3px; border-radius: 3px; background-color: rgba(0, 0, 0, 0.05);">id_asc</code>.</p></div></form><h3 style="font-family: &quot;Open Sans&quot;, &quot;Helvetica Neue&quot;, Helvetica, Arial, sans-serif; font-weight: 700; margin-right: 1136px; padding: 0px 28px; box-sizing: border-box; display: block; text-shadow: rgb(255, 255, 255) 0px 1px 0px; font-size: 15px; margin-top: 2.5em; margin-bottom: 0.8em; color: rgb(51, 51, 51); font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(234, 242, 246); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">Response Format</h3>

Field | Type
-- | --
cursor_string | CursorString
events | Event[]

</body>
</html>

[OSU WEB v2 Doku for get_events](https://osu.ppy.sh/docs/index.html#get-events)
### JSON Response example
[EVENTS_dump_notimpl_JSON.txt](https://github.com/NiceAesth/aiosu/files/12431833/EVENTS_dump_notimpl_JSON.txt)

![image](https://github.com/NiceAesth/aiosu/assets/13745823/610c30a2-73ec-40c9-ac99-a06a633db007)

**Comment by NiceAesth** [comment]
https://github.com/NiceAesth/aiosu/pull/169
